rust/stracciatella/tests/test_vfs.rs

```diff
@@ -80,25 +80,85 @@ mod vfs {
         temp.close().expect("close temp dir");
     }
 
+    #[test]
+    fn read_dir() {
+        let (temp, dir, dir_fs) = create_temp_dir();
+        create_foo_slf(&dir); // foo.slf
+        create_foobar_slf(&dir); // foobar.slf
+        create_file(&dir.join("foo/foo1.txt"));
+        create_file(&dir.join("foo/foo2.txt"));
+        create_file(&dir.join("foo/foo3.txt"));
+
+        let mut vfs = Vfs::new();
+        vfs.add_dir(&dir).expect("dir");
+        add_slf(&mut vfs, &dir_fs, "foo.slf");
+        add_slf(&mut vfs, &dir_fs, "foobar.slf");
+
+        let root_paths = ["foo", "foo.slf", "foobar.slf"];
+        let foo_paths = ["foo3.txt", "foo2.txt", "foo1.txt", "bar.txt", "bar"];
```

Comment:
Should we add a few test cases for non-ASCII directory names?

Reply:
Good call. I added some more tests and prompty found a bug in the SLF directory listing logic. Nevertheless that should be fixed now and tests for non-ASCII directory (and file) names are there.